src/components/Api.js

```diff
@@ -0,0 +1,204 @@
+  patchProfileAvatar(link) {
+    return fetch(this._baseUrl + "/users/me/avatar", {
+      method: "PATCH",
+      headers: {
+        authorization: this._headers.authorization,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        avatar: link,
+      }),
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+        return Promise.reject(`Error: ${res.status} + ${res.message}`);
+      })
+      .then((result) => {
+        return result;
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+
+  /*---------------------------------------------------*/
+  /*             Profile Api functions                 */
+  /*---------------------------------------------------*/
+
+  // – Get the current user’s info
+  // GET https://around-api.en.tripleten-services.com/v1/users/me
+  getProfile() {
+    return fetch(this._baseUrl + "/users/me", {
+      method: "GET",
+      headers: {
+        authorization: this._headers.authorization,
+      },
+    })
+      .then((res) => {
+        if (res.ok) return res.json();
+        return Promise.reject(`Error: ${res.status}`);
+      })
+      .then((result) => {
+        return result;
+        //console.log(result);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+
+  // – Update your profile information
+  // PATCH https://around-api.en.tripleten-services.com/v1/users/me
+  patchProfile(nameVar, bioVar) {
+    return fetch(this._baseUrl + "/users/me", {
+      method: "PATCH",
+      headers: {
+        authorization: this._headers.authorization,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        name: nameVar,
+        about: bioVar,
+      }),
+    })
+      .then((res) => {
+        if (res.ok) return res.json();
+        return Promise.reject(`Error: ${res.status}`);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+
+  /*---------------------------------------------------*/
+  /*             Card Api functions                    */
+  /*---------------------------------------------------*/
+
+  // – Get all cards
+  // GET https://around-api.en.tripleten-services.com/v1/cards
+  getCards() {
+    return fetch(this._baseUrl + "/cards", {
+      method: "GET",
+      headers: {
+        authorization: this._headers.authorization,
+        "Content-Type": "application/json",
+      },
+    })
+      .then((res) => {
+        if (res.ok) return res.json();
+        return Promise.reject(`Error: ${res.status}`);
+      })
+      .then((result) => {
+        // console.log(result);
+        const uniqueCardIds = new Set();
+        const uniqueCards = result.filter(({ name }) => {
+          if (!uniqueCardIds.has(name)) {
+            uniqueCardIds.add(name);
+            return true;
+          }
+          return false;
+        });
```

Comment:
Please, delete this filtering because `name` can be the same in many cards. You should render all cards from the server